helpers/calendar/analytics_helper.py

```diff
@@ -111,26 +111,28 @@ def get_all_events_in_a_room(self,
             - room_id - for specific room
             - event_start_time, event_end_time(Time range)
         """
-        events = EventsModel.query.all()
+        events = EventsModel.query.filter(
+            EventsModel.room_id == room_id,
+            EventsModel.state == 'active',
+            EventsModel.end_time < event_end_time,
+            EventsModel.start_time > event_start_time
+        )
         room_events = []
         for event in events:
```

Comment:
Great refactor @verenceLola! Can we use the `all()` method on `filter` so we return the filter result without having to loop through it again?

Reply:
I've been trying to get rid of the loop, a lot of areas to refactor, but I'll be done with it in a few minutes. @joshuaocero

Reply:
All done @joshuaocero .